torch/cuda/__init__.py

```diff
@@ -1211,23 +1211,18 @@ def _get_amdsmi_device_index(device: Optional[Union[int, Device]]) -> int:
 
 def _get_amdsmi_device_memory_used(device: Optional[Union[Device, int]] = None) -> int:
     handle = _get_amdsmi_handler()
-    device = _get_amdsmi_device_index(device)
     # amdsmi_get_gpu_vram_usage returns mem usage in megabytes
     mem_mega_bytes = amdsmi.amdsmi_get_gpu_vram_usage(handle)["vram_used"]
     mem_bytes = mem_mega_bytes * 1024 * 1024
     return mem_bytes
 
 
 def _get_amdsmi_memory_usage(device: Optional[Union[Device, int]] = None) -> int:
-    handle = _get_amdsmi_handler()
-    device = _get_amdsmi_device_index(device)
     handle = amdsmi.amdsmi_get_processor_handles()[device]
```

Comment:
This one doesn't like right? Is `amdsmi.amdsmi_get_processor_handles` really indexable by Union device?

Reply:
Yes, my mistake, have updated it, please check it again.